src/server/mod.rs

```diff
@@ -264,36 +250,37 @@ where
     }
 
     /// Start the server and listen for connections on the given address.
+    /// Blocks execution of current thread.
     ///
     /// # Example
     ///
     /// ```rust
     /// use libunftp::Server;
-    /// # use std::thread;
     ///
-    /// let mut server = Server::with_root("/srv/ftp");
-    /// # thread::spawn(move || {
-    /// server.listen("127.0.0.1:2000");
-    /// # });
+    /// let mut server = Server::with_root("/srv/ftp").listen("127.0.0.1:2000");
+    ///
+    /// // for simple use cases:
+    /// //tokio::run(server);
     /// ```
     ///
     /// # Panics
     ///
     /// This function panics when called with invalid addresses or when the process is unable to
     /// `bind()` to the address.
-    pub fn listen(self, addr: &str) {
+    pub fn listen<'a>(self, addr: &str) -> Box<dyn Future<Item = (), Error = ()> + Send + 'a> {
```

Comment:
Consider renaming `listen` now to somthing that doesn't hint that it blocks. Pehaps `listener`?

Reply:
nice find, fixed! thanks!